In [2]:
import pandas as pd
import numpy as np

In [3]:
titanic = pd.read_excel('titanic_M1.xlsx')
titanic

pclass  survived                                             name  \
0        1.0         1                    Allen, Miss. Elisabeth Walton   
1        1.0         1                   Allison, Master. Hudson Trevor   
2        1.0         0                     Allison, Miss. Helen Loraine   
3        1.0         0             Allison, Mr. Hudson Joshua Creighton   
4        1.0         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1214     3.0         0                              Smiljanic, Mr. Mile   
1215     3.0         0                                Smith, Mr. Thomas   
1216     3.0         1                               Smyth, Miss. Julia   
1217     3.0         0       Soholt, Mr. Peter Andreas Lauritz Andersen   
1218     3.0         0                    Somerton, Mr. Francis William   

         sex      age  sibsp  parch      ticket       fare    cabin embarked  \
0     female  29.0000      0      0       24160  2113375.0       B5        S   
1       male   0.9167      1      2      113781  1515500.0  C22 C26        S   
2     female   2.0000      1      2      113781  1515500.0  C22 C26        S   
3       male  30.0000      1      2      113781  1515500.0  C22 C26        S   
4     female  25.0000      1      2      113781  1515500.0  C22 C26        S   
...      ...      ...    ...    ...         ...        ...      ...      ...   
1214    male      NaN      0      0      315037    86625.0      NaN        S   
1215    male      NaN      0      0      384461    77500.0      NaN        Q   
1216   Mujer      NaN      0      0      335432    77333.0      NaN        Q   
1217    male  19.0000      0      0      348124    76500.0    F G73        S   
1218    male  30.0000      0      0  A.5. 18509    80500.0      NaN        S   

     boat   body                        home.dest  
0       2    NaN                     St Louis, MO  
1      11    NaN  Montreal, PQ / Chesterville, ON  
2     NaN    NaN  Montreal, PQ / Chesterville, ON  
3     NaN  135.0  Montreal, PQ / Chesterville, ON  
4     NaN    NaN  Montreal, PQ / Chesterville, ON  
...   ...    ...                              ...  
1214  NaN    NaN                              NaN  
1215  NaN    NaN                              NaN  
1216   13    NaN                              NaN  
1217  NaN    NaN                              NaN  
1218  NaN    NaN                              NaN  

[1219 rows x 14 columns]

a) Describa brevemente la cantidad de valores faltantes para cada una de la Base de Datos.

In [4]:
titanic.isna().sum()

pclass         76
survived        0
name            0
sex             0
age           239
sibsp           0
parch           0
ticket          0
fare            0
cabin         927
embarked        2
boat          751
body         1110
home.dest     475
dtype: int64

RESPUESTA: Faltantes Pclass:76, age:239, cabin:927, embarked:2, todos a completar.

b) En la columna Pclass, completar los valores NA, asignando a las personas a la clase 2.

In [7]:
titanic["pclass"].isna().sum()

np.int64(76)

In [9]:
titanic["pclass"] = titanic["pclass"].fillna(2)

c) En la clase “Sex”, imputar los datos de manera de poder completar la base solo con los valores “male” y “female”.

In [10]:
titanic["sex"].unique()

array(['female', 'male', 'Hombre', 'Mujer'], dtype=object)

In [11]:
titanic["sex"] = titanic["sex"].replace("Hombre", "male")
titanic["sex"] = titanic["sex"].replace("Mujer", "female")

In [12]:
titanic["sex"].unique()

array(['female', 'male'], dtype=object)

d) En los casos de las personas de las cuales no se tenga la edad, rellenar los datos con el promedio de edad de aquellos que sí tienen el dato.

In [13]:
titanic["age"].isna().sum()

np.int64(239)

In [17]:
titanic["age"].mean()

np.float64(30.026275510204076)

In [18]:
titanic["age"] = titanic["age"].replace(np.nan, titanic["age"].mean())

e) En la columna embarked, asignar el valor “S” a aquellos datos que se encuentran con valor vacío.

In [19]:
titanic["embarked"].isna().sum()

np.int64(2)

In [26]:
titanic["embarked"] = titanic["embarked"].fillna("S")

In [27]:
titanic

pclass  survived                                             name  \
0        1.0         1                    Allen, Miss. Elisabeth Walton   
1        1.0         1                   Allison, Master. Hudson Trevor   
2        1.0         0                     Allison, Miss. Helen Loraine   
3        1.0         0             Allison, Mr. Hudson Joshua Creighton   
4        1.0         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1214     3.0         0                              Smiljanic, Mr. Mile   
1215     3.0         0                                Smith, Mr. Thomas   
1216     3.0         1                               Smyth, Miss. Julia   
1217     3.0         0       Soholt, Mr. Peter Andreas Lauritz Andersen   
1218     3.0         0                    Somerton, Mr. Francis William   

         sex        age  sibsp  parch      ticket       fare    cabin  \
0     female  29.000000      0      0       24160  2113375.0       B5   
1       male   0.916700      1      2      113781  1515500.0  C22 C26   
2     female   2.000000      1      2      113781  1515500.0  C22 C26   
3       male  30.000000      1      2      113781  1515500.0  C22 C26   
4     female  25.000000      1      2      113781  1515500.0  C22 C26   
...      ...        ...    ...    ...         ...        ...      ...   
1214    male  30.026276      0      0      315037    86625.0      NaN   
1215    male  30.026276      0      0      384461    77500.0      NaN   
1216  female  30.026276      0      0      335432    77333.0      NaN   
1217    male  19.000000      0      0      348124    76500.0    F G73   
1218    male  30.000000      0      0  A.5. 18509    80500.0      NaN   

     embarked boat   body                        home.dest  
0           S    2    NaN                     St Louis, MO  
1           S   11    NaN  Montreal, PQ / Chesterville, ON  
2           S  NaN    NaN  Montreal, PQ / Chesterville, ON  
3           S  NaN  135.0  Montreal, PQ / Chesterville, ON  
4           S  NaN    NaN  Montreal, PQ / Chesterville, ON  
...       ...  ...    ...                              ...  
1214        S  NaN    NaN                              NaN  
1215        Q  NaN    NaN                              NaN  
1216        Q   13    NaN                              NaN  
1217        S  NaN    NaN                              NaN  
1218        S  NaN    NaN                              NaN  

[1219 rows x 14 columns]

In [28]:
titanic.isna().sum()

pclass          0
survived        0
name            0
sex             0
age             0
sibsp           0
parch           0
ticket          0
fare            0
cabin         927
embarked        0
boat          751
body         1110
home.dest     475
dtype: int64